x-pack/plugin/esql/qa/testFixtures/src/main/resources/keep.csv-spec

```diff
@@ -540,6 +540,41 @@ c:i
 1
 ;
 
+keepSubfieldViaSuffix
+required_capability: enrich_with_qualifier
+ROW a.b.c = "abc", b.c = "bc", x.y.z = "xyz"
+| EVAL xyz_implicit = z, xyz_implicit2 = y.z, xyz_full = x.y.z, bc_full = b.c, abc_full = a.b.c
+;
+
+a.b.c:keyword | b.c:keyword | x.y.z:keyword  | xyz_implicit:keyword | xyz_implicit2:keyword | xyz_full:keyword | bc_full:keyword | abc_full:keyword
+abc           | bc          | xyz            | xyz                  | xyz                   | xyz              | bc              | abc
+;
+
+// TODO: Test from index with deeper nesting levels.
+keepSubfieldViaSuffixFromIndex
+required_capability: enrich_with_qualifier
+FROM employees
+| WHERE emp_no == 10001
+| KEEP height.half_float
+| EVAL triple_height = half_float*3
+;
+
+height.half_float:double | triple_height:double
+             2.029296875 |          6.087890625
+;
+
+// TODO: For this to work we'll need to update our index resolution. Do we want to?
+keepSubfieldViaSuffixFromIndexDirectly-Ignore
+required_capability: enrich_with_qualifier
+FROM employees
+| WHERE emp_no == 10001
+| KEEP half_float
+;
+
+height.half_float:double
+             2.029296875
```

Comment:
After chatting with @luigidellaquila , we found out that we probably_have_ to make this work (i.e. make the index resolver ask for `*.field` in addition to every `field`).

Reason: otherwise, ambiguities might sometimes be resolved by accident.

```
FROM employees
| ENRICH some_policy AS p
| KEEP half_float
```
If `some_policy` has a field `half_float`, then this _should_ be ambiguous due to `height.half_float` from `employees`. But if we don't ask the index resolver for whether there's `*.half_float` in `employees`, this query will silently return `p.half_float` as if `height.half_float` didn't exist!